stock_quant_package_product_packaging/models/stock_quant_package.py

```diff
@@ -52,3 +64,35 @@ def auto_assign_packaging(self):
                 )
                 if packaging:
                     pack.write({"product_packaging_id": packaging.id})
+
+            if not pack.package_type_id:
+                # if no package type could be set by auto assign,
+                # fallback on the default product's package type (if any)
+                pack._sync_package_type_from_single_product()
+
+    def _sync_package_type_from_packaging(self):
+        for package in self:
+            if package.package_type_id:
+                # Do not set package type for delivery packages
+                # to not trigger constraint like height requirement
+                # (we are delivering them, not storing them)
+                continue
+            package_type = package.product_packaging_id.package_type_id
+            if not package_type:
+                continue
+            package.package_type_id = package_type
+
+    def _sync_package_type_from_single_product(self):
```

Comment:
@TDu will still need to call this method and give a product in argument as `package.single_product_id` won't beot yet computed